### PR TITLE
W9 - D1 - PhuTran4 - Unit test

### PR DESCRIPTION
**Pull request by at-phutran4**
unit test #55

### CODE REVIEW

File: app/src/testAt-phutran/java/asiantech/internship/summer/AccountValidationTest.java [hunk 1]
```diff
@@ -0,0 +1,181 @@
+package asiantech.internship.summer;
```
Comment (at-dev-support-01) [inline app/src/testAt-phutran/java/asiantech/internship/summer/AccountValidationTest.java]: WARNING: [TooManyStaticImports] Too many static imports may lead to messy code
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#TooManyStaticImports

Comment (at-dev-support-01) [inline app/src/testAt-phutran/java/asiantech/internship/summer/AccountValidationTest.java]: WARNING: [TooManyStaticImports] Too many static imports may lead to messy code
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#TooManyStaticImports

File: app/src/at-phutran/java/asiantech/internship/summer/unittest/UnitTestActivity.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_unit_test);
+        mEdtUsername = findViewById(R.id.edtUsername);
+        mEdtPassword = findViewById(R.id.edtPassword);
+        mTvResult = findViewById(R.id.tvResult);
+        mBtnLogin = findViewById(R.id.btnLogin);
+        mBtnLogin.setOnClickListener(v -> {
+            String username = mEdtUsername.getText().toString();
```
Comment (at-trungnguyen) [inline app/src/at-phutran/java/asiantech/internship/summer/unittest/UnitTestActivity.java]: `String username = mEdtUsername.getText().toString().trim();`

File: app/src/at-phutran/java/asiantech/internship/summer/unittest/UnitTestActivity.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            String password = mEdtPassword.getText().toString();
```
Comment (at-trungnguyen) [inline app/src/at-phutran/java/asiantech/internship/summer/unittest/UnitTestActivity.java]: `String password = mEdtPassword.getText().toString().trim();`

File: app/src/at-phutran/java/asiantech/internship/summer/unittest/UnitTestActivity.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            if (username.isEmpty() || password.isEmpty()) {
```
Comment (at-trungnguyen) [inline app/src/at-phutran/java/asiantech/internship/summer/unittest/UnitTestActivity.java]:
```
if (TextUtils.isEmpty(username) || TextUtils.isEmpty(password)) {
```

File: app/src/at-phutran/java/asiantech/internship/summer/unittest/UtilValidate.java [hunk 5]
```diff
@@ -0,0 +1,178 @@
+package asiantech.internship.summer.unittest;
+
+public class UtilValidate {
```
Comment (at-trungnguyen) [inline app/src/at-phutran/java/asiantech/internship/summer/unittest/UtilValidate.java]: `public final class UtilValidate {`

File: app/src/at-phutran/java/asiantech/internship/summer/unittest/UtilValidate.java [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+    public final static String USER = "trandinhphu";
+    public final static String USER_UPCASE = "TranDinhPhu";
+    public final static String USERNAME_RIGHT = "tranDinhPhu";
+
+    private UtilValidate() {
+    }
+
+    public static boolean isLengthUsernameRight(String username) {
+        return username.length() > 8 && username.length() < 21;
+    } // check length username
```
Comment (at-trungnguyen) [inline app/src/at-phutran/java/asiantech/internship/summer/unittest/UtilValidate.java]: wrong comment format, delete `// check length username`

File: app/src/at-phutran/java/asiantech/internship/summer/unittest/UtilValidate.java [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+    public final static String CHECK_LENGTH_PASSWORD = "Password must more than 8 characters";
+    public final static String CHARACRER_PASSWORD_REPEAT = "Password isn't loop 1 character too 2 times next";
+    public final static String END_PASSWORD = "Password don't finish with special character or digit";
+    public final static String UPPER_PASSWORD = "Password must have at least 3 non consecutive capital character";
+    public final static String SUCCESSFUL = "Successful!!!";
+    public final static String USER = "trandinhphu";
+    public final static String USER_UPCASE = "TranDinhPhu";
+    public final static String USERNAME_RIGHT = "tranDinhPhu";
+
+    private UtilValidate() {
```
Comment (at-trungnguyen) [inline app/src/at-phutran/java/asiantech/internship/summer/unittest/UtilValidate.java]:
```
private UtilValidate() {
    // no-op
}
```

File: app/src/at-phutran/java/asiantech/internship/summer/unittest/UtilValidate.java (hunk 6 above)
Comment (at-trungnguyen) [reply app/src/at-phutran/java/asiantech/internship/summer/unittest/UtilValidate.java]: if want to comment, press `ctr + enter` on function name. It is look like:
```
/**
     * This method use to check length of username
     *
     * @param username username
     * @return true if length > 8 and < 21
     */
    public static boolean isLengthUsernameRight(String username) {
        return username.length() > 8 && username.length() < 21;
    }
```

File: app/src/at-phutran/res/layout/activity_unit_test.xml [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+        android:id="@+id/btnLogin"
+        style="@style/MyButton"
+        android:layout_width="wrap_content"
+        android:layout_height="40dp"
+        android:layout_gravity="center_horizontal"
+        android:layout_margin="10dp"
+        android:background="@drawable/shapeimage1"
+        android:text="@string/login"
+        android:textColor="@color/colorWhite"
+        android:textSize="20sp" />
```
Comment (at-trungnguyen) [inline app/src/at-phutran/res/layout/activity_unit_test.xml]: convention is `dp`?

File: app/src/testAt-phutran/java/asiantech/internship/summer/AccountValidationTest.java [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+    @Test
+    public void testCaseUpcaseNotMoreTwoUpperCaseContinous() {
+        doReturn("TranDinhphu").when(mAccountPassword).getPassword();
+        assertFalse(UtilValidate.isUpcaseNotMoreTwoUpperCaseContinous(mAccountPassword.getPassword(), 3));
+        doReturn("trandinhphuUUU").when(mAccountPassword).getPassword();
+        assertFalse(UtilValidate.isUpcaseNotMoreTwoUpperCaseContinous(mAccountPassword.getPassword(), 3));
+    }
+
+    @Test
+    public void testCaseResultLogin() {
```
Comment (at-trungnguyen) [inline app/src/testAt-phutran/java/asiantech/internship/summer/AccountValidationTest.java]: big test case? can you separate it?

**Comment by at-trungnguyen** [comment]
@at-phutran4 Please check my comment!

File: app/src/at-phutran/java/asiantech/internship/summer/unittest/UtilValidate.java (hunk 6 above)
Comment (at-trungnguyen) [reply app/src/at-phutran/java/asiantech/internship/summer/unittest/UtilValidate.java]: @at-phutran4 please add comment for `public` method.

File: app/src/at-phutran/res/layout/activity_unit_test.xml (hunk 8 above)
Comment (at-trungnguyen) [reply app/src/at-phutran/res/layout/activity_unit_test.xml]: You need put `textSize` in `dimens.xml` file.

**Pull request merged**